x-pack/plugins/fleet/common/authz.ts

```diff
@@ -72,3 +86,70 @@ export const calculateAuthz = ({
     writeIntegrationPolicies: fleet.all && integrations.all,
   },
 });
+
+export function calculatePackagePrivilegesFromCapabilities(
+  capabilities: Capabilities | undefined
+): FleetAuthz['packagePrivileges'] {
+  if (!capabilities) {
+    return {};
+  }
+
+  const endpointActions = ENDPOINT_PRIVILEGES.reduce((acc, privilege) => {
+    return {
+      ...acc,
+      [privilege]: {
+        executePackageAction: capabilities.siem[privilege] || false,
```

Comment:
I think `siem` is a `Record` and `siem[privilege]` is a `boolean`

Reply:
yeah, that's correct. here's what `capabilities.siem` looks like:
```
{
    "show": true,
    "crud": true,
    "writeEndpointList": true,
    "readEndpointList": true,
    "writeTrustedApplications": true,
    "readTrustedApplications": true,
    "writeHostIsolationExceptions": true,
    "readHostIsolationExceptions": true,
    "writeBlocklist": true,
    "readBlocklist": true,
    "writeEventFilters": true,
    "readEventFilters": true,
    "writePolicyManagement": true,
    "readPolicyManagement": true,
    "writeActionsLogManagement": true,
    "readActionsLogManagement": true,
    "writeHostIsolation": true,
    "writeProcessOperations": true,
    "writeFileOperations": true
}
```

Reply:
Okay but that would mean `executePackageAction ` evaluates to `Record | boolean` instead of just `boolean`. Is that intended even though in the `FleetAuthz` type above it is defined as a `boolean`?  🤔